src/MSONNamedTypeParser.h

```diff
@@ -13,107 +13,112 @@
+    static MarkdownNodeIterator finalizeSignature(const MarkdownNodeIterator& node,
+        SectionParserData& pd,
+        const Signature& signature,
+        const ParseResultRef<mson::NamedType>& out)
+    {
 
-            if (pd.exportSourceMap()) {
+        mson::parseTypeName(signature.identifier, out.node.name, false);
+        mson::parseTypeDefinition(node, pd, signature.attributes, out.report, out.node.typeDefinition);
 
-                if (!out.node.name.empty()) {
-                    out.sourceMap.name.sourceMap = node->sourceMap;
-                }
+        if (pd.exportSourceMap()) {
 
-                if (!out.node.typeDefinition.empty()) {
-                    out.sourceMap.typeDefinition.sourceMap = node->sourceMap;
-                }
+            if (!out.node.name.empty()) {
+                out.sourceMap.name.sourceMap = node->sourceMap;
             }
 
-            // Named types should have type specification when sub-typed from primitive types
-            if (out.node.typeDefinition.baseType == mson::UndefinedBaseType) {
-                out.node.typeDefinition.baseType = mson::ImplicitObjectBaseType;
+            if (!out.node.typeDefinition.empty()) {
+                out.sourceMap.typeDefinition.sourceMap = node->sourceMap;
             }
+        }
 
-            // Setup named type context
-            pd.namedTypeContext = out.node.name.symbol.literal;
-
-            return ++MarkdownNodeIterator(node);
+        // Named types should have type specification when sub-typed from primitive types
+        if (out.node.typeDefinition.baseType == mson::UndefinedBaseType) {
+            out.node.typeDefinition.baseType = mson::ImplicitObjectBaseType;
         }
 
-        static MarkdownNodeIterator processDescription(const MarkdownNodeIterator& node,
-                                                       const MarkdownNodes& siblings,
-                                                       SectionParserData& pd,
-                                                       const ParseResultRef<mson::NamedType>& out) {
+        // Setup named type context
+        pd.namedTypeContext = out.node.name.symbol.literal;
 
-            return SectionProcessor<mson::ValueMember>::blockDescription(node, pd, out.node.sections, out.sourceMap.sections);
-        }
+        return ++MarkdownNodeIterator(node);
+    }
 
-        static MarkdownNodeIterator processNestedSection(const MarkdownNodeIterator& node,
-                                                         const MarkdownNodes& siblings,
-                                                         SectionParserData& pd,
-                                                         const ParseResultRef<mson::NamedType>& out) {
+    static MarkdownNodeIterator processDescription(const MarkdownNodeIterator& node,
+        const MarkdownNodes& siblings,
+        SectionParserData& pd,
+        const ParseResultRef<mson::NamedType>& out)
```

Comment:
This vs. line 111 https://github.com/apiaryio/snowcrash/pull/380/files#diff-fcd40d7ccfac0074f4dabdd5d0062a22R111

```
   static void finalize(      
        const MarkdownNodeIterator& node, SectionParserData& pd, const ParseResultRef<mson::NamedType>& out)
```